types/artifact.go

```diff
@@ -43,7 +43,8 @@ type Package struct {
 	Layer    Layer  `json:",omitempty"`
 
 	// Each package metadata have the file path, while the package from lock files does not have.
-	FilePath string `json:",omitempty"`
+	FilePath     string                  `json:",omitempty"`
+	Dependencies []godeptypes.Dependency `json:"Dependencies,omitempty"`
```

Comment:
I think it is `DependsOn []string`, no?

Reply:
You are correct it's more concise. But I can't build a dependency graph from it.

Reply:
Interesting. Why not?

Reply:
`PkgID` and list of dependent ids are stored. So it looks like everything is direct dependency.

Reply:
Currently, we have only direct dependencies in `Dependency`, right?
https://github.com/aquasecurity/go-dep-parser/blob/d316f5cc2cff93431f2ff8986ffbaf9a321da512/pkg/types/types.go#L13-L16

```
[]Dependency {
    {
        ID: "app1@1.0.0",
        DependsOn: {
            "app2@2.0.0",
            "app3@3.0.0",
        },
    },
    {
        ID: "app2@2.0.0",
        DependsOn: {
            "app4@4.0.0",
        },
    },
}
```

We just change it to

```
[]Package {
    {
        ID: "app1@1.0.0",
        Name: "app1",
        Version: "1.0.0",
        License: "MIT",
        DependsOn: {
            "app2@2.0.0",
            "app3@3.0.0",
        },
    },
    {
        ID: "app2@2.0.0",
        Name: "app2",
        Version: "2.0.0",
        License: "MIT",
        DependsOn: {
            "app4@4.0.0",
        },
    },
}
```

The information is the same, no?

Reply:
`app4@4.0.0` is an indirect dependency of `app1@1.0.0`. We can know it from `DependsOn`.